src/bin_grid.F90

```diff
@@ -166,6 +166,23 @@ integer function bin_grid_find(bin_grid, val)
 
   end function bin_grid_find
 
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+  !> Determines if a value is in a given bin.
+  logical function bin_grid_contains(bin_grid, i_bin, val)
+
+    !> Bin grid.
+    type(bin_grid_t), intent(in) :: bin_grid
+    !> Bin to see if it contains the value.
+    integer, intent(in) :: i_bin
+    !> Data value.
+    real(kind=dp), intent(in) :: val
+
+    bin_grid_contains = bin_grid%edges(i_bin) < val .and. &
+       bin_grid%edges(i_bin) < val
```

Comment:
Yes, good catch. While we are here, I'm changing the first comparison to `bin_grid%edges(i_bin) <= val` to catch values on that edge.

However this doesn't solve the case when we are in the final bin and the values is equal to the i_bin+1 edge. Should I add an extra bit of criteria if `i_bin = n_bin` to check to see if the value is equal to the last bin? Seems like a bit much but this would be more consistent with `linspace_find` and `logspace_find` type functions.

And I'll add some comment to describe this actual criteria.

Reply:
Good point. I think we want this code to be exactly consistent with the `_find()` variants.

Reply:
Alright - fixed this all up with https://github.com/compdyn/partmc/pull/168/commits/a5fb08eea19a4c532b4cb4f5b6884c16a71db9a5 and added some user error protection (invalid bin_grid and invalid i_bin that is outside the range of 1 to n_bin) in https://github.com/compdyn/partmc/pull/168/commits/8510d509f03a8e95970c430b62b511733f3faa55

Reply:
I think we still aren't correctly handling the cases when `val` is smaller than the first edge or larger than the last edge. In these cases `bin_grid_find()` returns `0` or `bin_grid_size()`, respectively. We should accept these values in `bin_grid_contains()` for consistency.

Reply:
Fixed in https://github.com/compdyn/partmc/pull/168/commits/3897117975c3ca480ae7de889e86672c56bf3a9a